<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Actions -->
    <param name="turn_to_human_action_name" type="string" value="/turn_to_human" />
    <param name="joy_priority_action" type="string" value="/joy_priority_action" />
    <param name="point_head_action" type="string" value="/head_controller/point_head_action" />
    <param name="disable_autohead_action" type="string" value="/pal_head_manager/disable" />
    <!-- Frames -->
    <!-- Default value, can be dynamically set from another node -->
    <param name="human_tf" type="string" value="speaker_sim" />
    <param name="robot_torso_tf" type="string" value="torso_lift_link" />
    <param name="robot_head_tf" type="string" value="head_2_link" />
    <param name="robot_base_tf" type="string" value="base_link" />
    <param name="point_head_tf" type="string" value="/head_controller/point_head_action/tilt_link" />
    <!-- Topics -->
    <param name="odometry_topic" type="string" value="/mobile_base_controller/odom" />
    <param name="joint_state_topic" type="string" value="/joint_states" />
    <param name="joy_priority_topic" type="string" value="/joy_priority" />
    <param name="command_velocity_topic" type="string" value="/joy_vel" />
    <!-- Constants -->
    <param name="use_joy_action" type="bool" value="true" />
    <param name="torso_rotation_velocity" type="double" value="0.8" />
    <param name="head_rotation_velocity" type="double" value="1.0" />
    <param name="max_head_rotation" type="double" value="1.3" />
    <!-- Nodes -->
    <node pkg="human_interactions" name="turn_to_human_action_server" type="turn_to_human_action_server.py" output="screen" />
</launch>
